### ISSUE TITLE
[tune] SigOptSearch suggester is not serialisable

### ISSUE THREAD

**Issue by Nintorac** [ray-project/ray#24864]
### What happened + What you expected to happen

I tried to run a ray tune job using the Sigopt suggester on a remote cluster. The sigopt suggester object was later found to be unserialisable however the stack trace gave no indication of this.

The stack trace looks like this

discussion around this issue can be found here https://ray-distributed.slack.com/archives/CNECXMW22/p1652417782100299

Thanks to Matthew Deng for finding the issue on this one!

### Versions / Dependencies

Python 3.8.12
ray==1.12.0

### Reproduction script

```
import ray
import numpy as np
import os
os.environ['SIGOPT_KEY'] = APIKEYHERE

from ray.tune.suggest.sigopt import SigOptSearch
from ray import tune
WORKING_DIR = os.getcwd()



def main():

	ray.init(
		address = "ray://127.0.0.1:10001",
		# address = "auto",
		runtime_env = {
			"working_dir": WORKING_DIR,
			"pip": ["sigopt==5.7.0"]
		}
	)
	
	n_observations = 20

	hyperparameter_space = [
          {
              'name': 'learning_rate',
              'type': 'double',
              'bounds': {
                  'max': np.log(0.01),
                  'min': np.log(0.0001)
              },
          },
          {
              'name': 'momentum',
              'type': 'double',
              'bounds': {
                  'min': 0.85,
                  'max': 0.99
              },
          },
      ]
	
	sigopt_search = SigOptSearch(
		# OmegaConf.to_container(config.search_space),
        hyperparameter_space,
		name="Tune distributed",
		max_concurrent=2, 
		observation_budget=n_observations,
		project="sigopt-ray-integration",
		metric=["val_loss"],
		mode=["min"]
		# metric=["val_loss", "training_loss"],
		# mode=["max", "min"]
	)

	tune_config = {
		# "config": config
	}
	analysis = tune.run(
		train_model,
		metric="val_loss",
		mode="min",
		config=tune_config,
		num_samples=n_observations,
		name="Tune distributed",
		resources_per_trial={'gpu': 1},
		search_alg=sigopt_search,
		# scheduler=FIFOScheduler(),
	)




def train_model(config):
    pass

main()

```

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

**Comment by cszhu** [comment]
Hello! This P2 issue has seen no activity in at least 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
